lib/rules/complexity.js

```diff
@@ -112,7 +119,8 @@ module.exports = {
             AssignmentPattern: increaseComplexity,
 
             // Avoid `default`
-            "SwitchCase[test]": increaseComplexity,
+            "SwitchCase[test]": () => IS_MODIFIED_COMPLEXITY || increaseComplexity(),
+            SwitchStatement: () => IS_MODIFIED_COMPLEXITY && increaseComplexity(),
```

Comment:
By default (classic cyclomatic complexity), empty switch statements and switch statements with only `default` clause count as 0, but with modified cyclomatic complexity they would count as 1?

Reply:
That's a very good question. Logic would suggest that if there is no branching at all, the complexity should be 0, but it seems like other analysis tools that support modified complexity always just give it a score of `1` without analyzing its contents. Here are my findings from the internet:

* I was able to verify that that's how it works in [lizard](https://github.com/terryyin/lizard/blob/master/lizard_ext/lizardmodified.py)
* I don't have access to Understand, but here's what their [docs](https://support.scitools.com/support/solutions/articles/70000582297-understanding-mccabe-cyclomatic-complexity) say:
> Modified Cyclomatic Complexity: Each decision in a multi-decision structure statement (switch in C/C++) is not counted and instead the entire structure counts as 1
(there's also more [detailed documentation](https://documentation.scitools.com/pdf/metricsdoc.pdf) in a PDF with an illustration, see page 53)
* I don't have access to MatLab, but there is a [detailed table](https://www.mathworks.com/help/matlab/matlab_prog/measure-code-complexity-using-cyclomatic-complexity.html) in its docs that describe how its `cyclomaticTest` module works, and I'm inferring from it that it also just gives the whole `switch` statement a score of 1.
* [Project Metrics](https://www.aivosto.com/project/help/pm-complexity.html)' docs also say "+1 for an entire Select Case"

It is a bit odd, and it wouldn't be hard to analyze the contents of switch in AST, but I think we should stay consistent with other implementations.

Reply:
Agreed.